bot/exts/moderation/watchchannels/_watchchannel.py

```diff
@@ -353,8 +353,16 @@ async def prepare_watched_users_data(
             watched_iter = reversed(watched_iter)
 
         list_data["info"] = {}
+
+        # Pre-fetch all watched users to avoid sequential API calls
+        # Filter out None to avoid AttributeError from disnake.utils.get()
+        members = await asyncio.gather(*[
+            get_or_fetch_member(ctx.guild, user_id)
+            for user_id, _ in watched_iter
+        ])
+        members = list(filter(None, members))
         for user_id, user_data in watched_iter:
-            member = await get_or_fetch_member(ctx.guild, user_id)
+            member = disnake.utils.get(members, id=user_id)
```

Comment:
You're already iterating over the entire list. You can build a dictionary instead and not have to iterate over all members in a loop.

Reply:
Do you mean something like this, instead of the filter?
`members = {member.id: member for member in members if member is not None}`

Reply:
yeah

Reply:
makes sense, pushed